generators/app/templates/src/app/components/CustomTextInput/index.js

```diff
@@ -12,36 +12,125 @@ import styles from './styles';
 // https://github.com/facebook/react-native/issues/5859
 
 class CustomTextInput extends PureComponent {
-  state = { showPassword: false };
+  state = {
+    showPassword: false,
+    focused: false,
+    value: this.props.value,
+    focusedAnim: new Animated.Value(this.props.value ? 1 : 0)
+  };
 
   handleShowPassword = () => {
     this.setState(prevState => ({ showPassword: !prevState.showPassword }));
   };
 
-  render() {
-    const placeholderTextColor = this.props.value ? transparent : this.props.placeholderTextColor;
+  toggle = isActive => {
+    const { animationDuration, easing, useNativeDriver } = this.props;
+    this.isActive = isActive;
+    Animated.timing(this.state.focusedAnim, {
+      toValue: isActive ? 1 : 0,
+      duration: animationDuration,
+      easing,
+      useNativeDriver
```

Comment:
Mmm, I think `useNativeDriver: true` does not support animations over positioning. If so, I'd declare this in `false` and remove the `prop`. For reference: https://facebook.github.io/react-native/docs/animations.html#caveats